secret/secret.go

```diff
@@ -0,0 +1,96 @@
+package secret
+
+import (
+	"log"
+	"os"
+
+	"github.com/SUSE/scf-secret-generator/model"
+	"github.com/SUSE/scf-secret-generator/password"
+	"github.com/SUSE/scf-secret-generator/ssh"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/rest"
+)
+
+// The name of the secret stored in the kube API
+const SECRET_NAME = "secret"
+
+func GetSecretInterface() corev1.SecretInterface {
+	// Set up access to the kube API
+	kubeConfig, err := rest.InClusterConfig()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	clientSet, err := kubernetes.NewForConfig(kubeConfig)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	return clientSet.CoreV1().Secrets(os.Getenv("KUBERNETES_NAMESPACE"))
+}
+
+func UpdateSecrets(s corev1.SecretInterface, secrets *v1.Secret, create, dirty bool) {
+	if create {
+		_, err := s.Create(secrets)
+		if err != nil {
+			log.Fatal(err)
+		}
+		log.Println("Created `secret`")
+	} else if dirty {
+		_, err := s.Update(secrets)
+		if err != nil {
+			log.Fatal(err)
+		}
+		log.Println("Updated `secret`")
+	}
+}
+
+func GetOrCreateSecrets(s corev1.SecretInterface) (create bool, secrets *v1.Secret) {
+	// check for existing secret, initialize a new Secret if not found
+	secrets, err := s.Get(SECRET_NAME, metav1.GetOptions{})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			log.Println("`secret` not found, creating")
+			create = true
+
+			secrets = &v1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: SECRET_NAME,
+				},
+				Data: map[string][]byte{},
+			}
+		} else {
+			log.Fatal(err)
+		}
+	}
+
+	return
+}
+
+func GenerateSecrets(manifest model.Manifest, secrets *v1.Secret) (dirty bool) {
+	sshKeys := make(map[string]ssh.SSHKey)
+
+	// go over the list of variables and run the appropriate generator function
+	for _, configVar := range manifest.Configuration.Variables {
+		if configVar.Secret && configVar.Generator != nil {
+			if configVar.Generator.Type == model.GeneratorTypePassword {
+				dirty = dirty || password.GeneratePassword(secrets.Data, configVar.Name)
```

Comment:
I'm so sorry; I don't know what I was thinking yesterday! ☹️ 

Your original logic was actually correct, and the change I asked for broke it: because of the short-circuiting behaviour of logical expressions, this will now only ever generate *one* new password at most, and skip all the rest. It has to be `... || dirty` to keep the proper record, but not interfere with generating additional passwords.

Reply:
Where did you branch to make your SSL changes? Any objection if I delete this commit and force push my branch?

Reply:
I'm always rebasing on top of your latest pushes (that's how I saw that you didn't run `dep ensure` because I ended up with your dependencies in my commit).

I'm fine with deleting that commit and doing a force-push; that is easy to fix during the rebase. I just didn't want the refactoring itself to be force-pushed, because that would be painful to rebase on top of.